### PR TITLE
Publish coverage

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/main/java/io/nats/client/impl/NatsJetStream.java [hunk 1]
```diff
@@ -369,9 +364,6 @@ NatsJetStreamSubscription createSubscription(String subject, String queueName,
                                                  PullSubscribeOptions pullSubscribeOptions) throws IOException, JetStreamApiException {
         // first things first...
         boolean isPullMode = pullSubscribeOptions != null;
-        if (handler != null && isPullMode) {
-            throw new IllegalStateException("Pull mode is not allowed with dispatcher.");
-        }
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsJetStream.java:367]: this condition is not possible anymore

File: src/main/java/io/nats/client/impl/NatsJetStream.java [hunk 2]
```diff
@@ -571,9 +563,6 @@ ConsumerInfo lookupConsumerInfo(String stream, String consumer) throws IOExcepti
     }
 
     private String lookupStreamBySubject(String subject) throws IOException, JetStreamApiException {
-        if (subject == null) {
-            throw new IllegalArgumentException("subject cannot be null.");
-        }
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsJetStream.java:576]: internal method, subject has already been validated

File: src/main/java/io/nats/client/impl/NatsPublishAck.java [hunk 3]
```diff
@@ -38,7 +38,7 @@ public NatsPublishAck(byte[] response) throws IOException {
         }
 
         if (hasError()) {
-            throw new IllegalStateException(getError());
+            throw new JetStreamApiException(this);
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsPublishAck.java:41]: This way the error contains the full api response object and is consistent with other errors

File: src/main/java/io/nats/client/PublishOptions.java [hunk 4]
```diff
@@ -204,7 +204,7 @@ public Builder expectedLastMsgId(String lastMsgId) {
          * @param sequence the expected last sequence number
          * @return builder
          */
-        public Builder expectedLastSeqence(long sequence) {
+        public Builder expectedLastSequence(long sequence) {
```
Comment (scottf) [inline src/main/java/io/nats/client/PublishOptions.java:207]: fixed spelling

File: src/test/java/io/nats/client/impl/JetStreamGeneralTests.java [hunk 5]
```diff
@@ -108,45 +108,6 @@ public void testJetStreamSubscribe() throws Exception {
         });
     }
 
-    @Test
-    public void testJetStreamPublish() throws Exception {
-        runInJsServer(nc -> {
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/JetStreamGeneralTests.java:113]: moved to a JetStreamPubTests

File: src/test/java/io/nats/client/impl/JetStreamPubTests.java [hunk 6]
```diff
@@ -13,41 +13,214 @@
...
+    private void assertContainsPublishAck(PublishAck pa, List<Long> seqnos) {
+        assertEquals(STREAM, pa.getStream());
+        assertFalse(pa.isDuplicate());
+        assertTrue(seqnos.contains(pa.getSeqno()));
+        seqnos.remove(pa.getSeqno());
+    }
+
+    @Test
+    public void testPublishExpectations() throws Exception {
+        runInJsServer(nc -> {
```
Comment (scottf) [inline src/test/java/io/nats/client/impl/JetStreamPubTests.java:182]: @ColinSullivan1 I had removed your original tests and finally put them back

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**